src/core/namespace.c

```diff
@@ -1391,14 +1416,9 @@ static int apply_one_mount(
                                                mount_entry_path(m));
                 }
 
-                if (geteuid() == 0)
-                        runtime_dir = "/run";
-                else {
-                        if (asprintf(&tmp, "/run/user/" UID_FMT, geteuid()) < 0)
-                                return -ENOMEM;
-
-                        runtime_dir = tmp;
-                }
+                runtime_dir = settle_runtime_dir(geteuid() == 0 ? RUNTIME_SCOPE_SYSTEM : RUNTIME_SCOPE_USER);
```

Comment:
so i i'd be ok with changing the parameter to simply `scope` here, i.e. actually always use the scope-specific inaccessible inode dir. this would mean that for root's user services overmounted device nodes would show up as socket inodes rather than inaccessible device nodes, but I think that's not too bad.

I'd also be fine to leave it as is, but then it should come with a comment, because it's not obvious:

/* We don't pass the literal runtime scope through here but one based purely on our UID. This means that the root user's --user services will use the host's inaccessible inodes rather then root's privat ones. This is preferable since it means device nodes that are overmounted to make them inaccessible will be overmounted with a device node, rather than an AF_UNIX socket inode */
```

or something like that.